Extensions/Firebase/tests/FirebaseExtension.spec.js

```diff
@@ -0,0 +1,226 @@
+/**
+ * A firebase configuaration of a project made only for those tests.
+ */
+const firebaseConfig = {
+  apiKey: 'AIzaSyBwPnGpfEBXDjwQrWfU0wqgp4m9qEt7YM8',
+  authDomain: 'gdtest-e11a5.firebaseapp.com',
+  databaseURL: 'https://gdtest-e11a5.firebaseio.com',
+  projectId: 'gdtest-e11a5',
+  storageBucket: 'gdtest-e11a5.appspot.com',
+  messagingSenderId: '254035412678',
+  appId: '1:254035412678:web:2ddd6b83019b7f259b79c7',
+  measurementId: 'G-4REML26L59',
+};
+
+/**
+ * Turns a callback variable into a promise.
+ * @param {(callbackVariable: {setString: (result: "ok" | string) => void}) => any} executor
+ * @returns
+ */
+const promisifyCallbackVariables = (executor) =>
+  new Promise((done, err) => {
+    const callbackResult = new gdjs.Variable();
+    executor(
+      {
+        setString: (string) => {
+          if (string === 'ok') done(callbackResult);
+          else err(string);
+        },
+      },
+      callbackResult
+    );
+  });
+
+/** A complex variable using all variables types. */
+const variable = new gdjs.Variable();
+gdjs.evtTools.network._objectToVariable(
+  { my: 'test', object: { with: 'all', types: 2 }, it: ['is', true] },
+  variable
+);
+
+// The tests require an internet connection, as a real Firebase instance is used.
+const describeIfOnline = navigator.onLine ? describe : describe.skip;
```

Comment:
I wonder if this'll be enough, having virtual box and other things installed can make navigator.onLine always return true.